src/context/toast/toast.test.tsx

```diff
@@ -206,6 +206,31 @@ describe("toast", () => {
       jest.useRealTimers();
     });
 
+    it("should not close on its own if shouldTimeout is false", async () => {
+      jest.useFakeTimers();
+      user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+      const { Component, hook } = renderComponentWithHook();
+      render(<Component />, {
+        wrapper,
+      });
+      act(() => {
+        hook.current.info("test string", true, { shouldTimeout: false });
+      });
+      expect(screen.getByDataCy("toast")).toBeInTheDocument();
+
+      // Advance timer, which would normally trigger timeout.
+      act(() => {
+        jest.advanceTimersByTime(TOAST_TIMEOUT);
+      });
+      await waitFor(() => {
+        expect(screen.getByDataCy("toast")).toBeInTheDocument();
+      });
```

Comment:
I'm wondering if this works without mocking the timers?

Reply:
Hmm it does, but test is supposed to test that the toast does not disappear even when the `setTimeout` time elapses. I tried doing it a different way by mocking `setTimeout` but that doesn't really work because I think the `useEffect` in toast actually gets called multiple times. In the end I just deleted the test